### DISCUSSION TITLE
Problems with Physics when use animation

### DISCUSSION THREAD

**Discussion by halx99** [axmolengine/axmol#1595] [Q&A] [answered]
### Discussed in https://github.com/axmolengine/axmol/discussions/1545

<div type='discussions-op-text'>

<sup>Originally posted by **paulocoutinhox** December 26, 2023</sup>
Hi,

I have a small problem when i use Physics with animation.

I made a small project to show:
https://github.com/paulocoutinhox/axmol-bomb-test

When i use this code:

```
// create animation
auto animation = Animation::createWithSpriteFrames(frames, 0.1f);
auto animate = Animate::create(animation);

auto bomb = Sprite::create();
bomb->runAction(RepeatForever::create(animate));

bomb->setName("shoot");
bomb->setContentSize(Vec2(78, 64));

// setup the physical body
auto physicsBody = PhysicsBody::createCircle(bomb->getContentSize().width / 2, PhysicsMaterial(1.0f, 0.1f, 0.8f));
physicsBody->setMass(1.0f);
bomb->setPhysicsBody(physicsBody);
```

The bomb goes to:

<img width="1238" alt="ss3" src="https://github.com/axmolengine/axmol/assets/395096/9b8497f9-2e73-4a16-93f3-fe3cde43ab05">

But when i use this code:

```
auto bomb = Sprite::create( "bomb.png");

bomb->setName("shoot");
bomb->setContentSize(Vec2(78, 64));

// setup the physical body
auto physicsBody = PhysicsBody::createCircle(bomb->getContentSize().width / 2, PhysicsMaterial(1.0f, 0.1f, 0.8f));
physicsBody->setMass(1.0f);
bomb->setPhysicsBody(physicsBody);
```

The bomb goes to down, correctly.

Thanks.</div>

**Comment by paulocoutinhox** [comment]
Hi,

I have updated the bomb-test project to create two sprites, one with animation and other without animation and left then fall with physics/gravity:
https://github.com/paulocoutinhox/axmol-bomb-test

I also make a video to ppl understand:

https://github.com/axmolengine/axmol/assets/395096/fe3bf581-8a59-4551-abac-02b511de56fd

**Comment by aismann** [comment]
As @rh101 has written:
The only thing I've noticed is that a person can get a lot more help if they have shown that a decent amount of effort has been put into tracking down the source of the issue (**simply debugging may not be enough**).

@paulocoutinhox 
I have made a small look on it, it needs more time.

It's only a known bug at this moment...may not easy to fix it in short time.
So please give some more help.

**Comment by rh101** [comment]
@paulocoutinhox If you printed out the PhysicsBody variables for each of the two sprites, such as position, rotation etc. etc., you would have noticed that nothing is different between the two objects, except the position.

You could have narrowed the issue down further by checking the `Animate` code to see what it actually does, and eventually you would have realised that the only lines that modify the sprite are these 2 lines:

```
static_cast<Sprite*>(_target)->setSpriteFrame(_origFrame);
static_cast<Sprite*>(_target)->setBlendFunc(blend);
```

`setBlendFunc` doesn't affect anything, so the only thing left to check is `setSpriteFrame`.  Placing a breakpoint in that method to trace through it and see what code is executed would lead you to the only line that would make sense to affect anything, being:
`setTextureRect(spriteFrame->getRect(), _rectRotated, spriteFrame->getOriginalSize());`

The contents of the `setTextureRect` is:
```
void Sprite::setTextureRect(const Rect& rect, bool rotated, const Vec2& untrimmedSize)
{
    _rectRotated = rotated;

    Node::setContentSize(untrimmedSize);
    _originalContentSize = untrimmedSize;

    setVertexRect(rect);
    updateStretchFactor();
    updatePoly();
}
```

For a quick test, one could simply comment out the contents to see what happens, and funny enough, things start working.  Uncommenting lines until the issue re-appears would reveal this one line as the culprit:
`Node::setContentSize(untrimmedSize);`

Putting a breakpoint on that call shows the `untrimmedSize` as `97.5, 80.0`.

Now, a few things to note:

1 - The frame sizes are all `78,64`, as shown in the PLIST file.
2 - Your own code has this call: `bomb->setContentSize(Vec2(78, 64));`
3 - They don't match the `untrimmedSize` being set, implying that some kind of content scaling is happening, most likely due to `Director::setContentScaleFactor`

If you comment out the hard-coded line, `bomb->setContentSize(Vec2(78, 64));`, then everything starts working, although your sprite without animation is huge and incorrect:
![image](https://github.com/axmolengine/axmol/assets/8603230/9fab1702-d62f-433e-8f62-da28c2d4e3f9)

That is because of this line: `bomb->setTexture("bomb.png");`.  You're using the entire sprite sheet for the image texture, which is strange.  It would have made more sense to call this: `bomb->setSpriteFrame("frame1.png");`, but that would require fixing other issues in your code, such as ensuring that the sprite sheet is loaded only once, and not just for the animated sprite. 

So, move this code to `MainScene::init()`, which is where it belongs:
```
auto spriteFrameCache = SpriteFrameCache::getInstance();
spriteFrameCache->addSpriteFramesWithFile("animation.plist", "bomb.png");
```

To unload it, you can do so in the destructor, `MainScene::~MainScene`, or when you're done with it (meaning no sprites references those textures).

Now, knowing that removing the call `bomb->setContentSize(Vec2(78, 64));` fixes the issue means something else is happening.

Remember the `untrimmedSize` as `97.5, 80.0`, well that means things are being scaled.  So, based on that, testing with a content scale factor to 1.0, via `director->setContentScaleFactor(1.0f);`,  then everything also works correctly, regardless of whether the call to `bomb->setContentSize(Vec2(78, 64));` is made or not.

The content scale factor of 1 would mean the `untrimmedSize` is always `78,64`, so the call to `Node::setContentSize(untrimmedSize);` doesn't actually do anything, given the original size is the same as the new size:

```
void Node::setContentSize(const Vec2& size)
{
    if (!size.equals(_contentSize))
    {
        _contentSize = size;

        _anchorPointInPoints.set(_contentSize.width * _anchorPoint.x, _contentSize.height * _anchorPoint.y);
        _transformUpdated = _transformDirty = _inverseDirty = _contentSizeDirty = true;
    }
}
```

The only thing I can think of is that changing the size of the sprite after the physics body has been created is somehow affecting the transforms, which are used in calculations within the PhysicsBody.  

So, in summary, what works is any of the following:

Delete the line `bomb->setContentSize(Vec2(78, 64));`

or

Set it to the correct scaled size (being `97.5,80`) via:  `bomb->setContentSize(Vec2(78, 64) / AX_CONTENT_SCALE_FACTOR());`

or 

Only set the content size of the sprite if and only if the sprite is not animating, so calling `bomb->setContentSize(Vec2(78, 64));` on the non-animating sprite is fine, since the size never changes.

or 

Use a content scale factor of 1.0 via `director->setContentScaleFactor(1.0f);`

**The most important thing to remember is**: The content size of a sprite cannot be changed after the physics object is set on a node.  This means you should not call `setContentSize()` with a different size (directly or indirectly) *after* you have set the physics body on the sprite.  You can scale the sprite, rotate it, or anything else, just do not call `setContentSize` once a physics body is attached.  This also means sprite frames in an animation must all be the same size.

In the end, this entire issue can be reproduced with a few lines that have nothing to do with `Animate`:
```
auto sprite = Sprite::create("SomeImage.png");
auto physicsBody = PhysicsBody::createCircle(sprite->getContentSize().width / 2, PhysicsMaterial(1.0f, 0.1f, 0.8f));
physicsBody->setMass(1.0f);
sprite->setPhysicsBody(physicsBody);
sprite->runAction(Sequence::createWithTwoActions(DelayTime::create(1.5f), CallFunc::create([sprite]() {
    sprite->setContentSize(sprite->getContentSize() / 2); // Problem will show itself after 1.5 seconds, when setContentSize is called with a different size
})));
```

**Comment by aismann** [comment]
Added a wiki hint

**Comment by paulocoutinhox** [comment]
Hi,

In my code don't have anything changing content size after set physics body. I don't know if you see my code, but i will paste it here:

```
Sprite *MainScene::createBomb(bool animated)
{
    auto bomb = Sprite::create();
    
    if (animated)
    {
        // create animation
        auto spriteFrameCache = SpriteFrameCache::getInstance();
        spriteFrameCache->addSpriteFramesWithFile("animation.plist", "bomb.png");

        // create sprites
        Vector<SpriteFrame *> frames;
        char frameName[100];

        for (int i = 1; i <= 4; i++)
        {
            sprintf(frameName, "frame%d.png", i);

            SpriteFrame *frame = spriteFrameCache->getSpriteFrameByName(frameName);

            if (frame)
            {
                frames.pushBack(frame);
            }
        }

        // create animation
        auto animation = Animation::createWithSpriteFrames(frames, 0.1f);
        auto animate = Animate::create(animation);

        bomb->runAction(RepeatForever::create(animate));
    }
    else
    {
        bomb->setTexture("bomb.png");
    }

    bomb->setName("shoot");
    bomb->setContentSize(Vec2(78, 64));

    // setup the physical body
    auto physicsBody = PhysicsBody::createCircle(bomb->getContentSize().width / 2, PhysicsMaterial(1.0f, 0.1f, 0.8f));
    physicsBody->setMass(1.0f);
    bomb->setPhysicsBody(physicsBody);

    return bomb;
}
```

It create a animated or not sprite, and at last, it add physics body.

The line that change something on it and make it work is related to scale as you write above, not content size position in my code, since it is before physics body definition:

```
bomb->setContentSize(Vec2(78, 64) / AX_CONTENT_SCALE_FACTOR());
or
bomb->setContentSize(Vec2(78, 64) / _director->getContentScaleFactor());
```

Thanks.

**Comment by rh101** [comment]
> In my code don't have anything changing content size after set physics body. I don't know if you see my code, but i will paste it here:

As already mentioned in my previous post, `setContentSize` cannot be called after the physics body is set, whether it is from your own code or indirectly through other code paths.  In your case, it is done so via `setSpriteFrame`, which is called by via the `Animate` action, as I explained in detail in my previous post.

> The line that change something on it and make it work is related to scale as you write above, not content size position in my code, since it is before physics body definition:

That is not correct. It is definitely `setContentSize` that is causing the issue, and nothing to do with scaling the object.  Just because the content scale factor is used doesn't mean the object is being scaled. You're just setting the correct size by using it, which means when the animation is playing, the size is exactly the same, so all calls to `setContentSize` made during the animation will not change anything, because of this `if (!size.equals(_contentSize))` check in `Node::setContentSize`.

If that is still not completely clear, just re-read the post I made above, and follow the code path yourself to see what actually happens.

**Comment by paulocoutinhox** [comment]
Hi, relax man, i read it, all lines.

1. I only set the same sprite to not animated object ONLY to use the same image in my test, i already know that it is not beautiful, but it is only a test.
2. I already understand that cant change content size after add physic body.

What i do to solve is set content size with original frame size and change only the scale:

```
bomb->setName("shoot");
bomb->setContentSize(Vec2(78, 64) / AX_CONTENT_SCALE_FACTOR());
bomb->setScale(0.5);
```

This is wrong?

**Comment by rh101** [comment, marked as answer]
> This is wrong?

Sorry, I'll clarify. This code is correct:
`bomb->setContentSize(Vec2(78, 64) / AX_CONTENT_SCALE_FACTOR());`

What is not correct is this remark regarding the content size and your usage of it:
> The line that change something on it and make it work is related to scale as you write above, not content size position in my code, since it is before physics body definition:

The reason is that the line `bomb->setContentSize(Vec2(78, 64)` was setting a content size of `78,64`, but when it is animating,  `setSpriteFrame` was calling `setContentSize` with `97.5, 80.0`, because of the scale factor.  As a result of this, whatever parameters are changed within the node end up affecting the physics object.  That is why the fix is to divide by the content scale factor when setting the content size, since future calls to the `setContentSize` by the Animate code will no longer trigger changes in that node.

**Comment by paulocoutinhox** [comment]
Nice. Thanks.

**Comment by paulocoutinhox** [comment]
rsrsrsrs

[Animation](https://devxz.com/sprite-sheet-animator/?sW=78&sH=64&fC=4&fT=100&c=white&zoom=1&fS=0&pT=0&pL=0&spH=0&spV=0&b=true&img=data%3Aimage%2Fpng%3Bbase64%2CiVBORw0KGgoAAAANSUhEUgAAATgAAABACAYAAACOeWtyAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAElBJREFUeNrsnQuUHGWVgG9V9fs90%2FOMIeSBTILIwwBJSDQQPOgiURaOsIhnicYVeSyes5x4UFc8ssd1V3dZFhSIHJBFICLGdY2L7JoNcjaRREACqwmTZCaQTJJ59fT7WV3Ve29N90w%2Fqntmema6qnr%2Be85NTaqr67%2F19f%2Ffe%2F%2F6%2F6qfy%2BVy0MwytsI25%2Bds7UsBEyZM9C8mhoAJCwZMmIMzuLhva5n4O7o96MHNOahDuP%2FUdL6P32G1hQkLCMzB6VfQSdlx823UL6Ha8%2FuO4mYn6s9Q%2F4AOr1qf%2Fc9ROdSfs2DAggETYwSBBePgsNE5cPMb1Mt5Dw%2BWi23AOXmQhrLvz%2FZm7pVj8r342Sk87iVydKjDqGbUlajXoV6A%2BlEWDFgwYAHBOMItoEGG76Dey7cI4PiUGzi3B7juiwBC70IudBKyAyJkj2Yge0KEXHqcCd8qgBySAGTlv9S4z0PNNms3oohVtWAAGAwAgwEdQ413qmDwP83c5SrwIgenFhDydWbKgIDfnQgIzcarmNFcBYEFlcHd%2F8DDNT%2B%2Fa7LBKn9a19qB71oOps%2F8AsDZAZCTQNr9dTBx%2Fwams8zKwblkTqluclCCxC%2Bjyi7Uu8m5LZDA9w3FuZUFAzMFg9UTweB9GAy2YjDYWiUYvMJ6B6x3oHWvYKF0UVejukDgFCfGX3bHuHNTclgBhI3fAPngM4qzU3bZOcWlpfcni7O%2FlxYIKxYMWEBomiDQ9A7O3582P9phWnujS8Cmlw8OZkcZBQvWOB5AkiZ2pX%2BfBGlYaaNPoP7tAmqsLBiwgNA0QaBpHdw377l7A24%2Bz3Hcdc9b21tuhDEFlDQqAffmU8D3XAsgWJRj5TfQh0nixHczb6Qgc1Dp5z9ClZfS5oVyI%2FinMYkFAxYQmiYIzNrBda1YqSm5O%2B68s9yxbaZNa1v76kvXrYfzL%2F4QeD1eCD3zD%2BCLjEEGG6PQ9hpkn7oKuGWbIDfWB7n%2BPeOOLiZD%2Bn8TNNBAtXEbOrZ%2F0fv1z1YG%2B94Zb3gcp3C7vLN19Y0Qb1gwMDK%2FQ%2BTRI9JHt3q0Cwh653dI4yDQNBkcOrYe3DzW3tl1xaaPXwOrPnghRCMROPj6ATj8f2%2FD26Hh5PZ2k51S4eSLMbCu6wU%2BcEz5Lo0OiofTIB7NUMM%2BgbtuxYb624WQfqBjK%2BF23vkXQBCDQUs0OOfBoJlkqL%2B3x4%2FcNnS3X7EVIqx3UFs846lZ44OAqoNbt2HttE%2BwrNtXsW%2F%2FvgNn4eZK1EtQqQEtRW1DdcL4SEgMdRT1XdRe1NdRX167fs3JWmXheem7n4Dxm41Eazc6thepopjN5n%2B86ppr7Ws2bISx0RH4%2BXM%2Fhj%2B99WZYkqRn8fMXfgawFx0c3ci8H50cl31BHI8UyDSXUcBTT%2F%2FJfGMN1QNTjcVMpJHcUFW57dzxDBw6Ppp4pE1wNCoYGJnb2vUfgTEMCK2xUMMDArXT2bBrBDcMArkveoVj3%2FGbNAkCqvPgqjm4Ze9fWvVEO556fglVdtS%2F4AXhPJfHB75WHzgcdrBaLWA2CcBzHDpqDmQ5BzKWK4pZSKXSEItGsXEFIJlIUEb7E9Snb95y03tl5ye3%2F7zZarvW5nIr%2B5LRCGQz6ZMdXd1n3XTrVvB4vbD7xV3w2u%2F2jsqyTGnt44HlVqUTX5iHg6A24uYrqFfk7w2czFfcB%2FGYt6tdX7V5OMVdhGqVzaDcvkzBIF%2FRZxQMZjJnifgZndvSt38XxYCgHGRabMaAYFdukE83INQ7x8so7TSN3A525%2BKdAuckPvZrXMC1nVMWBHI1g0C9jGbt4PCCLsTN13leuMHp9fEdXR3gdTvBYhZm7G3TGRECgRCcGRiQxUyG5sZ8G%2B17C7tRBO2Xdrf3Kk9bJ%2FWroJDyhkcGoburC6782Mfh33c8mwuOBbbjJ1%2F91j8%2FpDS8u36wrVaRVip2pnbOhYPTKzd0bKH5CgZz4eB0zO2uRgQEo7bTjdHT8KhnPEtrZBCo28EhMLpL%2BF1eMP2lu9XPdXd3gsdlU7x%2FsQiCAHanHex2G5gtZjCZTRgd%2BPHUVJYhi9FBRGDJZAqScex3Yz%2BcokZgLAwn%2Bo%2FnEOBzeOgShPZhT3uXql3h4TOQjkdFvJYb0LHtKv5sCgdXl8zGwemVG%2BquKZ4bnHUwmI2D0ys3dGy7GhkQjNxOt8QHYJszC4WSGxEE6hpkQGifRu%2F8Q4fH5%2BtctAjavA4ExJccY0NQ3hYP2DH15cpgFkMltdqs4PK4lH2JeALCwYjyA7T6LuQGTg3eMnjqFFgczqr2eDu6ITQom9OJGFWwEgf3%2FTu%2FV%2FNa7vubv27YnVYjcVML3FrdodY7t6LG9wpMztUqCQhaDCTojdtTzsVwYGwAvuUSj59v4Tpzydy0g0Ddg2gzyeD27ztAdwUfNJktt3s7F0FXewu4HZZSj4me39%2FuB4fTPrHPgRd92erLYNXK8%2BCc5SvA4%2FaC0%2BmECPbNI5EIHOs7Cn889Ed44%2BDrkE6n8wCTEBgJKJEjHE1AX%2B9REKw2cPs7VH8IKZuF0RN9dL%2FNj1mcOF0As3Vw08ngjMIN64LY6EZYK4MzCjfM4uaM21xkcAbixs9Hr6AuB4fQyHXvtDpcV7d0ITS%2FG2yW0gTQhX16f4d%2FIrX1%2BXzwqU9cBxs3bASbzT6lMbF4DHbv%2BQ386qVfQSIRV1LjwHAAYtE4pDJZOPLOMcgiIAxHwPF8xfdH3jsGsiR1o4Mb1IuDMxI3rAuDenFwjFt9Ds5I3NDBDc53EJiWg0NolEruxv71Ol9nN3S3uipuTtJITIt%2FsqJ%2BeP1H4NZbtoDD7pjxBVC02P7ko%2FDmW28q%2Fw8GQhAaC0FGlODokX5IpZLQ0r0YI8QkPFnKIrg%2BGtZu1UsGZzRuesngGLf6HJzRuHUu7xELE83nS6Z0cPl099cIbRPde%2Bj2u8BaAc2L0CZfh7Jp41XwhS1%2FVXIMldPXfwwO9x6GoeEhiKPnd7vc0NXZBR9Y9QE4e8nSiuP%2F6V%2B%2FWwQviPDCkEZ4vRQhxAz4uhZPpMHh4dOQikUfQOd2z0wAzJeDMxo3%2FN49WjTScgfHuNXn4IzGDZ2bwm2%2BHdx0BhkexHR3E42M%2BD22CmiU7hZDI1l98SUlAF5%2BZQ%2F8x3%2F%2BAkZGR6oW8r5Fi%2BH6T14P69ZcPu55Ecilqy%2BbAEdlUD8fMA1etmIZHMO%2BfjQwDDQcHRkZJGg0mearoB9h3Bg3xk1jbjUdHEaFmwSz%2BXaKCE6bGdwOa%2BWNSuzLl8t%2F7f419JzbA8FgEH704yfg0DuHpjTk1OkBePixh2Dvq3thC6bMFqtV6ecXC5VFEw7dmE0vOvtsOHHsCATPnIRMMkHQNmP2ltFDTTMaN6zcjFsd3DALYdx0zq1qFxWhLUL3%2FKfWRUt8VpsdFne4QSi7Ydi5qLNkFKYRQqM2Q6eHQJJzcHxgGEYH3sti%2BFm5dv2avj%2B7%2FpYZn2%2Buu6hG5Pbq3v2aNc5CF9WI3I6fCc17F2uqLqpRuRVkvvnxNT77Hs2fMVtt4HVZK6DR%2FJlGQLt8zXrY%2FvDj8P0HHoELzr9AKZPKFngO2lq9gDZSFnq%2FjroKjBvjxrjphBtfJeX9EC%2BYbna1tCnPpHmc1opjaHLgvPefTSb44udvU25ytra0wpe23lFSNtnl8bcD2Uo266CrwLgxboybjrhVy%2BDuc%2FpaOJrD4rZb1B%2FrcMx%2FVOBAfWY1lU02kF1upw2cvlY68D4dRFPGjXFj3HTErcLBcRx3Ns8Lm%2B3u8aF7l91Sabiz%2BmMdcyliVoTHf7QdYrEojAZG4bEnHinYqNhAQj%2Bs3ePD6CBs%2FuY9dy%2FTqqYN9fcalhvuZ9zq4Ia2a8bNyO20kdzURlE%2Fa3N7eIoKJoFXfdsAPZDbKNm3f5%2BiajbEIjHFPnrFi83l4RPh4Oc0jKyMG%2BPGuOmMm1oX9RY0Ytw4q%2FosEnrbgNZSbAPZmbf5Jg1NYtwYN8ZNZ9z4srT3LEwhV9GIDInVrA7OZNb%2BTedm8yQ4spNsRtvPxW7qcg26WYbnhr99w7k1Q31j3Orjhm2mIdzKM7hNlqJn0swm9TEInuc1B8fxXIWdeduv1MAcxo1xY9x0yK2cwCVm6%2BSoi0nQHtC0okTezrztF2lgAuPGuDFuOuRWnsOeK5gtRRFAfQQmJ%2BeAEzhNYZEN5VEib3tPI19i2UzcNDCHcWPcGprBrTAV9Zn5KkPMoiiC1iJmsxV25m1froE5jBvjxrjpkFu5g2vnhcmkTlZ5TlUxOqMDcOlMhZ15230amMO4MW6Mmw65lTs4T%2FEbOGVZHRwtPKG1pIpsKKTBedvdGpjDuDFujJsOudW8O5mVZHVw8STkqkSNhvTrsWx6W8GEnXlwWtrEuDFujJv%2BuJU7uEhOlov6z%2BrgaMmwZCKpGTgqm2yYtFPKRwjF3pgGJjFujBvjpkNu5Q5uhN6ZXpC0KFX9Ii0ZppWUl53KFMAp26AGJjFujBvjpkNu5Q6uL1s08pKqcZOS%2BtbF6WejhMpMld1bKNiZt71fg9%2BScWPcGDcdcit3cEckMVOS%2BmZqRAdaD1GW5YZBU5YnwzKLhewrpOh523s1qHCMG%2BPGuOmQW7mDe11Ml3r7WLL669NpcQlaD7FRQmUpC1oUSbTIvrztBzWocIwb48a46ZBbuYPbk0kmKgyTa4x60GKvtB7ifAuVQWWVRAq0q%2FiHzdv%2BsgYVjnFj3Bg3HXIrcXC5XO6kLEmHxXSqKN3MQSSernkSWux1PuEVFpQtF7KrMAeIbEbbj%2BI1NPyeSOfyHsatDmH1beFywzajyT04kmdTsdLRj3AsrayOMxW8kcGROe3r07nonGrQyB6yqyB5m58H7YRxY9wYN51xq%2Bbg5OJ5NpRiBsKJKU9GqenpE6fnZNSGzkHnKk93J%2Fr5aE8hJaeJg2Qz%2FvmkxhWOcWPcGDcdceNV0t93MYXclYyWeuN4SoRoYur1WkUxq6yHeGZgEC8%2BUQewhPJdOodYdqNy4n4D2kH2FCQZCVHauwttP65VbcOUm3FbYNzQds24GbmdNpJbtVd%2B%2Fl08FPyk3e3jip95C0SSyrvVrSrvfy8XmgNDSqvqFNZINFstYDIJEy%2Fio9Q2m5WUB3KV%2BTo08zkr1TwvTWokOyZ%2BaDxHPDRGIUIPa1Uybowb46YjbqoObu36NW%2Fs33dgRyw4%2Bhm3v6M4asDQWAy6Wl2qi1yoCT2qEY3EFJ2t0FwaKr%2F4WbZYMIBRIbsDbf6D1rWNcVtY3IpXaGfc9Mmt1sP22xKRUCibTlfcNDyDxqcy2Yb%2BmFQelVt8E5VsS0SCRGsb6EcYN8aNcdMJN75GdDiNLvj28PBpyJWNuNCQ7%2BBYvObkwrkU6stTecWvhSGbyDayUbFVJ8K4MW6Mm3648VPA%2B0lWzDwWHj6jdpMTRkIJGA7G0VvPz2MgkiQr5x8NJypesUI2oW3byUbQmTBujBvjpg9u01mt4svpRGxPZGRQ9UMaJRkYjkIolqo5k3omQueh8w2MREtGYQpCtpBN%2BOfdoF9h3Bg3xk1jblM6OPS8lN9uTkbDr1aDRxcajKbg5HAExiLJmg%2F%2B1hL6Hn2fzkPnU%2FshyAayhWzK26ZLYdwYN8ZNe26macJL7N934Go0eKcsSVd7O7qBU1lzkfre4XhaUVrKjFayVhZ7NfHKkmG0qg4tPEFA6PXFIqa29IaBtJiFVDqr%2FL%2BajPflz1BE%2BG%2F87w1kE%2BhcGDfGjXHTlhun9vrgdRvWqh6M8Gi9r4dMZsttHoRXWFl7voWeX4vk%2B%2FKU7laLCK%2Fu3d9wgF0rVk78vazbx7jVwc%2Bo3Gi6w2DfO5qxM2o7LZ4mMt%2F8uHrej85x3Kfxnx86PC0%2BV4tfNUrMhVA0oPkzyhBzLndb5%2FKen5Yfo2UFK3dwU8lQf69uuBmJnx65GYGfnrk1il9dV4xO8QX8pycRHnt69OTxXCIcrBiini0wOuf4uceeprJmAk2vgtfAuDFujJvWXdQZZnMX4eZrPC%2FcYHN5eJvLDWabve4Ul942kIpGZFmWduKuv0dgB40aQafISjTlZlR%2BeuFmNH5646brLqoaYAS4BDe3ot7MC8Iqi92BfX87mMwW4M1mwH3Ac%2Fln23Iy5CQJJFGk%2Frryhk96CZ4sSXS1z6E%2BjcDea8YGqlLxNOHWBAFCU24GDhC64GY4B6cC8UrUS1HPRV2K2obqyh9CD7yNor6LegT1NdSXEdaJZq9g06h8DeHWTPy04NYM%2FLTkZmgH10hpJgfH%2BDF%2BjJ%2FOHBwTJkyY6FX%2BX4ABAG2G9gwWlVSBAAAAAElFTkSuQmCC)

**Comment by aismann** [comment]
@paulocoutinhox 

> rsrsrsrs
> 
> [Animation]

What is the reason to put it here?

Please close this issue if  is "fixed".

**Comment by paulocoutinhox** [comment]
I can't, it was not open by me
